dsc/workflows/base/__init__.py

```diff
@@ -260,19 +255,36 @@ def submit_items(self, collection_handle: str) -> list:
             f"for batch '{self.batch_id}'"
         )
 
+        batch_metadata = {
+            self.get_item_identifier(item_metadata): item_metadata
+            for item_metadata in self.item_metadata_iter()
+        }
+
         items = []
-        for item_submission in self.item_submissions_iter():
+        for item_submission_record in ItemSubmissionDB.get_batch_items(self.batch_id):
```

Comment:
Hmm, I think this was briefly mentioned in the first part of [this comment](https://github.com/MITLibraries/dspace-submission-composer/pull/189#discussion_r2240783951), but it doesn't seem like [`ItemSubmissionDB.get_batch_items()` ](https://github.com/MITLibraries/dspace-submission-composer/blob/IN-1356-update-submit/dsc/db/models.py#L198-L205) does a lot beyond materializing all the elements in the generator. I propose:

a. Just using `ItemSubmissionDB.query` or
b. Update `ItemSubmissionDB.get_batch_items` to simply:
```python
return cls.query(batch_id)
```

I understand that having a `get_batch_items` method allows the method call to be more detailed explicit, but again, it really doesn't seem to be doing much (so perhaps **b** is a good compromise)? 🤔